src/js/pose_viewer/src/components/pose-viewer/pose-viewer.tsx

```diff
@@ -86,14 +87,30 @@ export class PoseViewer {
     this.resizeObserver.observe(this.element);
   }
 
-  private async getRemotePose() {
-    // Abort previous request
+  private async loadPose() {
+    // Abort previous request if it exists
     if (this.fetchAbortController) {
       this.fetchAbortController.abort();
     }
 
-    this.fetchAbortController = new AbortController();
-    this.pose = await Pose.fromRemote(this.src, this.fetchAbortController);
+    if (this.src instanceof Buffer) {
```

Comment:
here, maybe a small refactor:

```ts
    if(typeof this.src === 'string') {
      const isRemoteUrl = this.src.startsWith('http') || this.src.startsWith('//');
      const isBrowserEnvironment = typeof window !== 'undefined';

      if (isRemoteUrl || isBrowserEnvironment) {
        // Remote URL or Browser environment
        this.fetchAbortController = new AbortController();
        this.pose = await Pose.fromRemote(this.src, this.fetchAbortController);
      } else {
        // Local environment
        this.pose = await Pose.fromLocal(this.src);
      }
    } else {
      this.pose = Pose.from(this.src);
    }
```

That moves the responsibility for error/type checking to the `from` method,